File: frontend/src/types/session.ts
import type { Block } from "./blocks";

export interface PlanNode {
  id: string;
  type: "profession" | "module" | "topic" | "lesson";
  title: string;
  status: "planned" | "generating" | "ready";
  children: PlanNode[];
  lesson_id?: string;
}

export interface Classification {
  scale: "lesson" | "topic" | "module" | "profession";
  title: string;
  time_estimate: string;
}

export type SessionStatus =
  | "idle"
  | "classifying"
  | "scaffolding"
  | "generating"
  | "done"
  | "error";

export interface SessionState {
  sessionId: string | null;
  status: SessionStatus;
  classification: Classification | null;
  outline: PlanNode[] | null;
  currentLessonId: string | null;
  lessons: Record<string, Block[]>;
  errors: string[];
}

export interface SessionSnapshot {
  session_id: string;
  status: string;
  classification: Classification | null;
  outline: PlanNode[] | null;
  lessons: Record<
    string,
    { title: string; status: string; blocks: Block[] }
  >;
}

File: backend/app/schemas/session.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class PlanNode(BaseModel):
    """A node in the content plan tree."""

    id: str
    type: Literal["profession", "module", "topic", "lesson"]
    title: str
    status: Literal["planned", "generating", "ready"] = "planned"
    children: list[PlanNode] = []
    lesson_id: str | None = None


class SessionCreate(BaseModel):
    """Request body for creating a new generation session."""

    user_request: str


class SessionResponse(BaseModel):
    """Response for session creation."""

    session_id: str
